www/js/app.js

```diff
@@ -496,6 +496,60 @@ document.getElementById('disableDragAndDropCheck').addEventListener('change', fu
         }
     });
 });
+// Handle switching from jQuery to serviceWorker modes.
+document.getElementById('serviceworkerModeRadio').addEventListener('click', async function() {
+    document.getElementById('enableSourceVerificationCheckBox').style.display = '';
+    params.sourceVerification = getSetting('sourceVerification') === null ? true : getSetting('sourceVerification');
+    if (selectedArchive.isReady() && !(settingsStore.getItem("trustedZimFiles").includes(selectedArchive.file.name))) {
+        const response = await uiUtil.systemAlert('Is the loaded ZIM archive from a trusted source?\n If not, you can still read the ZIM file in Safe Mode (aka JQuery mode). Closing this window also opens the file in Safe Mode. This option can be disabled in Expert Settings', 'Security alert!', true, 'Open in Safe Mode', 'Trust Source');
+            console.log(response);
+            if (response) {
+                params.contentInjectionMode = 'serviceworker';
+                var trustedZimFiles = settingsStore.getItem('trustedZimFiles');
+                var updatedTrustedZimFiles = trustedZimFiles + archive.file.name + '|';
+                settingsStore.setItem('trustedZimFiles', updatedTrustedZimFiles, Infinity);
+            } else {
+                params.contentInjectionMode = 'jquery';
+                document.getElementById('jqueryModeRadio').checked = true;
+            }
+    }
+});
+document.getElementById('jqueryModeRadio').addEventListener('click', function() {
+    if (this.checked) {
+        document.getElementById('enableSourceVerificationCheckBox').style.display = 'none';
+    }
+});
+// Handle switching to serviceWorkerLocal mode for chrome-extension
+document.getElementById('serviceworkerLocalModeRadio').addEventListener('click', async function() {
+    document.getElementById('enableSourceVerificationCheckBox').style.display = ''; 
+    params.sourceVerification = getSetting('sourceVerification') === null ? true : getSetting('sourceVerification');
```

Comment:
My IDE says getSetting is undefined.

Reply:
the line wasn't doing anything, I added it by accident.